fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -210,7 +210,7 @@ def fit(self, X, y, **kwargs):
             if self.subsample != None:
                 current_estimator.fit(X_sub, y_reduction_sub)
             else:
-                current_estimator.fit(X, y_reduction, **{self.sample_weight_name: weights})
+                current_estimator.fit(X, y_reduction, **{self.sample_weight_name: weights.fillna(0)})
```

Comment:
Something is off when weights take NaN values. Are the labels or sensitive features undefined?